app/services/post_creation.rb

```diff
@@ -1,19 +1,21 @@
 # frozen_string_literal: true
 
 class PostCreation
-  def initialize(user, post_params)
+  def initialize(user:, post_params:)
     @user = user
     @post_params = post_params
   end
 
   def self.call(user, post_params)
-    new(user, post_params)
+    new(user: user, post_params: post_params).call
   end
 
   def call
     post = Post.new(@post_params.merge(user: @user))
-    return post if post.save
-
-    false
+    if post.save
+      true
+    else
+      false
```

Comment:
how would you show validation error messages to user if you return only boolean?
it should return hash { status: , record: } and depends on error or not return values